File: jiejiechen-react/src/models/enquiry-form.ts
export interface EnquiryFormElements extends HTMLFormControlsCollection {
  email: HTMLInputElement;
  firstName: HTMLInputElement;
  lastName: HTMLInputElement;
  phoneNumber: HTMLInputElement;
  company: HTMLInputElement;
}

export interface EnquiryForm extends HTMLFormElement {
  readonly elements: EnquiryFormElements;
}

File: services/DemoApi.Domain/EnquiryForm.cs
namespace DemoApi.Domain;

public record EnquiryForm
{
    public Int64 Id { get; set; }
    public string? Email { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? Phone { get; set; }
    public string? Company { get; set; }
    public DateTime TimeSubmitted { get; set; }
}